scripts/release/mirror.js

```diff
@@ -0,0 +1,361 @@
+      newNotes.push(notes2);
+    } else {
+      newNotes.push(...notes2);
+    }
+  }
+
+  newNotes = newNotes.filter((item, pos) => newNotes.indexOf(item) == pos);
+
+  if (newNotes.length == 0) {
+    return null;
+  }
+  if (newNotes.length == 1) {
+    return newNotes[0];
+  }
+
+  return newNotes;
+};
+
+/**
+ * @param {string|string[]|null} notes
+ * @param {RegExp} regex
+ * @param {string} replace
+ * @param {Function} versionMapper - Receives the source browser version and returns the target browser version.
+ * @returns {string|string[]|null}
+ */
+const updateNotes = (notes, regex, replace, versionMapper) => {
+  if (!notes) {
+    return null;
+  }
+
+  if (Array.isArray(notes)) {
+    return notes.map((note) =>
+      updateNotes(note, regex, replace, versionMapper),
+    );
+  }
+
+  return notes
+    .replace(regex, replace)
+    .replace(
+      new RegExp(`(?:${replace}|version)\\s(\\d+)`),
+      (match, p1) => replace + ' ' + versionMapper(p1),
+    );
+};
+
+/**
+ * @param {SupportStatement} data
+ * @returns {SupportStatement}
+ */
+const copyStatement = (data) => {
+  let newData = {};
+  for (let i in data) {
+    newData[i] = data[i];
+  }
+
+  return newData;
+};
+
+/**
+ * @param {...SupportStatement} data
+ * @returns {SupportStatement}
+ */
+const combineStatements = (...data) => {
+  const ignoredKeys = ['version_added', 'notes'];
+
+  const flattenedData = data.flat(2);
+  const sections = {};
+  let newData = [];
+
+  for (const d of flattenedData) {
+    const key = Object.keys(d)
+      .filter((k) => !ignoredKeys.includes(k))
+      .join('');
+    if (!(key in sections)) sections[key] = [];
+    sections[key].push(d);
+  }
+
+  for (const k of Object.keys(sections)) {
+    const currentStatement = sections[k][0];
+
+    if (sections[k].length == 1) {
+      newData.push(currentStatement);
+      continue;
+    }
+
+    for (const i in sections[k]) {
+      if (i === 0) continue;
+      const newStatement = sections[k][i];
+
+      const currentVA = currentStatement.version_added;
+      const newVA = newStatement.version_added;
+
+      if (newVA === false) {
+        // Ignore statements with version_added being false
+        continue;
+      } else if (typeof newVA === 'string') {
+        if (
+          typeof currentVA !== 'string' ||
+          compareVersions.compare(
+            currentVA.replace('≤', ''),
+            newVA.replace('≤', ''),
+            '>',
+          )
+        ) {
+          currentStatement.version_added = newVA;
+        }
+      }
+
+      const newNotes = combineNotes(currentStatement.notes, newStatement.notes);
+      if (newNotes) currentStatement.notes = newNotes;
+    }
+
+    if ('notes' in currentStatement && !currentStatement.notes) {
+      delete currentStatement.notes;
+    }
+    newData.push(currentStatement);
+  }
+
+  if (newData.length === 1) {
+    return newData[0];
+  }
+
+  // Remove duplicate statements and statements that are only version_added = false
```

Comment:
It looks like there's nothing in `combineStatements` that handles the case where the derived `version_added` and `version_removed` are the same version, which can happen. Such a statement should be removed, and this seems like the place to do it.